plaso/parsers/santa.py

```diff
@@ -0,0 +1,397 @@
+    certificate_hash (str): SHA256 hash for the certificate associated with the
+        executed process.
+    certificate_common_name (str): certificate common name.
+    pid (str): process id for the process.
+    ppid (str): parent process id for the executed process.
+    uid (str): user id associated with the executed process.
+    user (str): user name associated with the executed process.
+    gid (str): group id associated with the executed process.
+    group (str): group name associated with the executed process.
+    mode (str): Santa execution mode, for example Monitor or Lockdown.
+    process_path (str): process file path.
+    process_arguments (str): executed process with its arguments.
+  """
+
+  DATA_TYPE = 'santa:execution'
+
+  def __init__(self):
+    """Initializes event data."""
+    super(SantaExecutionEventData, self).__init__(data_type=self.DATA_TYPE)
+    self.action = None
+    self.decision = None
+    self.reason = None
+    self.process_hash = None
+    self.certificate_hash = None
+    self.certificate_common_name = None
+    self.quarantine_url = None
+    self.pid = None
+    self.ppid = None
+    self.uid = None
+    self.user = None
+    self.gid = None
+    self.group = None
+    self.mode = None
+    self.process_path = None
+    self.process_arguments = None
+
+
+class SantaFileSystemEventData(events.EventData):
+  """Santa file system event data.
+
+  Attributes:
+    action (str): event type recorded by Santa.
+    file_path (str): file path and name for WRITE/DELETE events.
+    file_new_path (str): new file path and name for RENAME events.
+    pid (str): process id for the process.
+    ppid (str): parent process id for the executed process.
+    process (str): process name.
+    process_path (str): process file path.
+    uid (str): user id associated with the executed process.
+    user (str): user name associated with the executed process.
+    gid (str): group id associated with the executed process.
+    group (str): group name associated with the executed process.
+  """
+
+  DATA_TYPE = 'santa:file_system_event'
+
+  def __init__(self):
+    """Initializes event data."""
+    super(SantaFileSystemEventData, self).__init__(data_type=self.DATA_TYPE)
+    self.action = None
+    self.file_path = None
+    self.file_new_path = None
+    self.pid = None
+    self.ppid = None
+    self.process = None
+    self.process_path = None
+    self.uid = None
+    self.user = None
+    self.gid = None
+    self.group = None
+
+
+class SantaMountEventData(events.EventData):
+  """Santa mount event data.
+
+  Attributes:
+    action (str): event type recorded by Santa.
+    mount (str): disk mount point.
+    volume (str): disk volume name.
+    bsd_name (str): disk BSD name.
+    fs (str): disk volume kind.
+    model (str): disk model.
+    serial (str): disk serial.
+    bus (str): device protocol.
+    dmg_path (str): DMG file path.
+    appearance (str): disk appearance date.
```

Comment:
This looks like a separate (and very useful!) timestamp. What do you think about adding this as a separate event as well?

Reply:
I agree, but I am not 100% sure the field is consistent across all DISKAPPEAR events; if appearance field is present we can create another SantaMountEventData() event with definitions.TIME_DESCRIPTION_FIRST_CONNECTED.